### PR TITLE
Feature/redux store

### PR DESCRIPTION
**Pull request by MLucky518**
Created basic redux skeleton and established a store with one reducer. Dispatch called from store in api helper function and sets profile data into state to help with searching

### CODE REVIEW

File: src/components/pages/ProfileList/RenderProfileListPage.js [hunk 1]
```diff
@@ -2,23 +2,34 @@ import React from 'react';
...
-        </figcaption>
-      </figure>
-    ))}
-  </div>
-);
+import { connect } from 'react-redux';
 
-export default RenderProfileListPage;
+function RenderProfileListPage(props) {
+  console.log(props.profiles);
```
Comment (tlewandowski18) [inline src/components/pages/ProfileList/RenderProfileListPage.js:8]: Remove console.log

File: src/api/index.js [hunk 2]
```diff
@@ -40,7 +41,10 @@ const apiAuthGet = authHeader => {
 
 const getProfileData = authState => {
   try {
-    return apiAuthGet(getAuthHeader(authState)).then(response => response.data);
+    return apiAuthGet(getAuthHeader(authState)).then(response => {
+      store.dispatch(setProfilesToState(response.data));
```
Comment (tlewandowski18) [inline src/api/index.js:45]: Good use of hooks

File: src/index.js [hunk 3]
```diff
@@ -17,13 +17,21 @@ import { ProfileListPage } from './components/pages/ProfileList';
 import { LoginPage } from './components/pages/Login';
 import { config } from './utils/oktaConfig';
 import { LoadingComponent } from './components/common';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import { Provider } from 'react-redux';
+import { appReducer } from './state/reducers/appReducer';
+
+export const store = createStore(appReducer, applyMiddleware(thunk));
```
Comment (tlewandowski18) [inline src/index.js:25]: Good use of middleware

File: src/state/actions/index.js [hunk 4]
```diff
@@ -1,5 +1,9 @@
-// import all of your actions into this file, and export them back out. 
-// This allows for the simplification of flow when importing actions into your components throughout your app.
-// Actions should be focused to a single purpose. 
-// You can have multiple action creators per file if it makes sense to the purpose those action creators are serving. 
-// Declare action TYPES at the top of the file
\ No newline at end of file
+export const POPULATE = 'POPULATE';
+
+export const setProfilesToState = data => {
```
Comment (tlewandowski18) [inline src/state/actions/index.js:3]: Action looks good

File: src/state/reducers/index.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
-// import all of your reducers into this file, and export them back out. 
-// This allows for the simplification of flow when importing reducers into your actions throughout your app.
-
+import { combineReducers } from 'redux';
+import { appReducer } from './appReducer';
 
+export default combineReducers({
+  appReducer,
+});
```
Comment (tlewandowski18) [inline src/state/reducers/index.js:6]: reducer looks good.  May need to add more specificity for data.

File: src/index.js [hunk 6]
```diff
@@ -51,8 +58,9 @@ function App() {
           component={() => <HomePage LoadingComponent={LoadingComponent} />}
         />
         <SecureRoute path="/example-list" component={ExampleListPage} />
-        
-        <SecureRoute path="/profile-list" component={ProfileListPage} /><Route component={NotFoundPage} />
+
+        <SecureRoute path="/profile-list" component={ProfileListPage} />
+        <Route component={NotFoundPage} />
```
Comment (wilcoxva) [inline src/index.js:63]: Nice change

**Review by wilcoxva — COMMENTED** [review]
Great overall!

**Pull request merged**